### ISSUE TITLE
docker-compose version 2

### ISSUE THREAD

**Issue by e-jigsaw** [sameersbn/docker-gitlab#630]
Hi

I tried to run docker-gitlab with docker-compose version 2.

`docker-compose.yml` is

```
version: '2'
services:
  postgresql:
    restart: always
    image: sameersbn/postgresql:9.4-13
    environment:
      - DB_USER=gitlab
      - DB_PASS=password
      - DB_NAME=gitlabhq_production
    volumes:
      - /srv/docker/gitlab/postgresql:/var/lib/postgresql
  gitlab:
    restart: always
    image: sameersbn/gitlab:8.5.4
    links:
      - redis:redisio
      - postgresql:postgresql
    ports:
      - "10080:80"
      - "10022:22"
    environment:
      - DEBUG=false
      - TZ=Asia/Kolkata
      - GITLAB_TIMEZONE=Kolkata

      - GITLAB_SECRETS_DB_KEY_BASE=long-and-random-alphanumeric-string

      - GITLAB_HOST=localhost
      - GITLAB_PORT=10080
      - GITLAB_SSH_PORT=10022
      - GITLAB_RELATIVE_URL_ROOT=

      - GITLAB_NOTIFY_ON_BROKEN_BUILDS=true
      - GITLAB_NOTIFY_PUSHER=false

      - GITLAB_EMAIL=notifications@example.com
      - GITLAB_EMAIL_REPLY_TO=noreply@example.com
      - GITLAB_INCOMING_EMAIL_ADDRESS=reply@example.com

      - GITLAB_BACKUP_SCHEDULE=daily
      - GITLAB_BACKUP_TIME=01:00

      - SMTP_ENABLED=false
      - SMTP_DOMAIN=www.example.com
      - SMTP_HOST=smtp.gmail.com
      - SMTP_PORT=587
      - SMTP_USER=mailer@example.com
      - SMTP_PASS=password
      - SMTP_STARTTLS=true
      - SMTP_AUTHENTICATION=login

      - IMAP_ENABLED=false
      - IMAP_HOST=imap.gmail.com
      - IMAP_PORT=993
      - IMAP_USER=mailer@example.com
      - IMAP_PASS=password
      - IMAP_SSL=true
      - IMAP_STARTTLS=false
    volumes:
      - /srv/docker/gitlab/gitlab:/home/git/data
  redis:
    restart: always
    image: sameersbn/redis:latest
    volumes:
      - /srv/docker/gitlab/redis:/var/lib/redis
```

And `docker-compose up` failed.

```
...
gitlab_1     | Initializing logdir...
gitlab_1     | Initializing datadir...
gitlab_1     | Installing configuration templates...
gitlab_1     | Configuring gitlab...
gitlab_1     | Configuring gitlab::database
gitlab_1     | ERROR:
gitlab_1     |   Please configure the database connection.
gitlab_1     |   Refer http://git.io/wkYhyA for more information.
gitlab_1     |   Cannot continue without a database. Aborting...
gitlab_1     |
...
```

Version 1 style `docker-compose.yml` is works, but version 2 did not works.

How to fix this problem?

Thanks

**Comment by e-jigsaw** [comment]
UPDATE:

I added below environment values. So I can start container but I had unicorn error.

```
      - DB_HOST=postgresql
      - DB_POST=5432
      - DB_USER=gitlab
      - DB_PASS=password
      - REDIS_HOST=redisio
      - REDIS_PORT=6379
```

```
...
gitlab_1 | 2016-03-08 01:32:24,105 INFO exited: unicorn (exit status 1; not expected)
gitlab_1 | 2016-03-08 01:32:25,110 INFO spawned: 'unicorn' with pid 586
gitlab_1 | 2016-03-08 01:32:26,112 INFO success: unicorn entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
gitlab_1 | 2016-03-08 01:32:26,548 INFO exited: unicorn (exit status 1; not expected)
gitlab_1 | 2016-03-08 01:32:27,552 INFO spawned: 'unicorn' with pid 589
gitlab_1 | 2016-03-08 01:32:28,554 INFO success: unicorn entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
gitlab_1 | 2016-03-08 01:32:29,222 INFO exited: unicorn (exit status 1; not expected)
gitlab_1 | 2016-03-08 01:32:30,226 INFO spawned: 'unicorn' with pid 592
...
```

**Comment by e-jigsaw** [comment]
I read #492 #311 and fix that error.

**Comment by frenchbeard** [comment]
Yep, as mentionned in the docs, v2 removes part of the environment variables created when linking containers. If it were to be completely optimized for use with compose v2, the gitlab container should rely on the hostname associated with the containers, not their IP (as you will see if you "exec" in the running gitlab container, with a simple `cat /etc/hosts`).
I was wondering if we could make the container support that form as well, checking for a "postgresql" and "redisio" in the hosts' file and default ports if none provided.

**Comment by 18601673727** [comment]
@e-jigasaw I don't think ur way solved this issue, in fact, configure DB parameters in gitlab service block just making the gitlab running with the "External PostgreSQL Server" not the container one so I'm trying to use the "network alias" way but not done yet.

**Comment by 18601673727** [comment]
I just submitted a new PR against this issue, so this could been closed maybe.

**Comment by metal3d** [comment]
Another solution is to set a .env file and give it to the three services

**Comment by metal3d** [comment]
To complete my answer, here is my configuration (local test):

```yaml
version: '2'
services:
    postgresql:
      image: sameersbn/postgresql:9.4-20
      env_file: .env
      volumes:
        - ./postgresql:/var/lib/postgresql
    gitlab:
      image: sameersbn/gitlab:8.7.0
      links:
        - redis:redisio
        - postgresql:postgresql
      ports:
        - "10080:80"
        - "10022:22"
      env_file: .env
      volumes:
        - ./gitlab:/home/git/data
    redis:
      env_file: .env
      image: sameersbn/redis:latest
      volumes:
        - ./redis:/var/lib/redis
```

And the .env is:

```bash
DEBUG=false
TZ=Asia/Kolkata
GITLAB_TIMEZONE=Kolkata
GITLAB_SECRETS_DB_KEY_BASE=761oGrTj5qg5TG41NM1UpafPl6q7pbU1
GITLAB_HOST=localhost
GITLAB_PORT=10080
GITLAB_SSH_PORT=10022
GITLAB_RELATIVE_URL_ROOT=

DB_HOST=postgresql
DB_POST=5432
DB_USER=gitlab
DB_PASS=password
DB_NAME=gitlabhq_production
DB_EXTENSION=pg_trgm

REDIS_HOST=redisio
REDIS_PORT=6379
```

**Comment by SolidNerd** [comment]
I will close this because #658 is merged and #661 is closed as well.
